pkg/state/state.go

```diff
@@ -1635,6 +1639,12 @@ func (st *HelmState) flagsForUpgrade(helm helmexec.Interface, release *ReleaseSp
 		flags = append(flags, "--cleanup-on-fail")
 	}
 
+	if helm.IsVersionAtLeast(3, 2) &&
+		(release.CreateNamespace != nil && *release.CreateNamespace ||
+			release.CreateNamespace == nil && (st.HelmDefaults.CreateNamespace == nil || *st.HelmDefaults.CreateNamespace)) {
+		flags = append(flags, "--create-namespace")
+	}
```

Comment:
Can we add an error message for the user who tried to use createNamespace with a version of Helm that doesn't support it? Probably like this:

```suggestion
	if (release.CreateNamespace != nil && *release.CreateNamespace ||
		release.CreateNamespace == nil && (st.HelmDefaults.CreateNamespace == nil || *st.HelmDefaults.CreateNamespace)) {
		if !helm.IsVersionAtLeast(3, 2) {
			return nil, fmt.Errorf("releases[].createNamespace requires Helm 3.2.0 or greater")
		}
		flags = append(flags, "--create-namespace")
	}
```